packages/core/src/routes/hook.ts

```diff
@@ -39,8 +47,24 @@ export default function hookRoutes<T extends AuthedRouter>(
         query: { includeExecutionStats },
       } = ctx.guard;
 
-      const hooks = await findAllHooks();
+      const { limit, offset, disabled: isPaginationDisabled } = ctx.pagination;
 
+      if (isPaginationDisabled) {
+        const hooks = await findAllHooks();
+
+        ctx.body = yes(includeExecutionStats)
+          ? await Promise.all(hooks.map(async (hook) => attachExecutionStatsToHook(hook)))
+          : hooks;
+
+        return next();
+      }
+
+      const [{ count }, hooks] = await Promise.all([
+        findTotalNumberOfHooks(),
+        findAllHooks(limit, offset),
```

Comment:
nit: the function name may not be appropriate

Reply:
Just aligned with other APIs. If we have a better name, I will rename them all.
And it's not a good idea to rename them all in this PR since that will cause so many files to change.